src/bridges/yearn/YearnBridge.sol

```diff
@@ -0,0 +1,205 @@
+     * the user deposits the underlying (or want) token as input (ex: DAI). When withdrawing from a Yearn vault, the user asks to withdraws some shares of the
+     * Yearn Vaults, where input is the yvToken (ex: yvDAI) and receive the underlying token as output (ex: DAI).
+     * @param _inputAssetA - ERC20 token to deposit or Yearn Vault ERC20 token to withdraw
+     * @param _outputAssetA - Yearn Vault ERC20 token to receive on deposit or ERC20 token to receive on withdraw
+     * @param _inputValue - Amount to deposit or withdraw
+     * @param _interactionNonce - Unique identifier for this DeFi interaction
+     * @param _auxData - Special value to indicate a deposit (0) or a withdraw (1)
+     * @return outputValueA - the amount of output asset to return
+     */
+    function convert(
+        AztecTypes.AztecAsset memory _inputAssetA,
+        AztecTypes.AztecAsset memory,
+        AztecTypes.AztecAsset memory _outputAssetA,
+        AztecTypes.AztecAsset memory,
+        uint256 _inputValue,
+        uint256 _interactionNonce,
+        uint64 _auxData,
+        address
+    )
+        external
+        payable
+        override(BridgeBase)
+        onlyRollup
+        returns (
+            uint256 outputValueA,
+            uint256,
+            bool
+        )
+    {
+        if (_inputValue == 0) {
+            revert ErrorLib.InvalidInputAmount();
+        }
+        if (_auxData == 0) {
+            if (_outputAssetA.assetType != AztecTypes.AztecAssetType.ERC20) {
+                revert ErrorLib.InvalidOutputA();
+            }
+            if (_inputAssetA.assetType == AztecTypes.AztecAssetType.ETH) {
+                _zapETH(msg.value, _outputAssetA);
+            }
+            outputValueA = IYearnVault(_outputAssetA.erc20Address).deposit(_inputValue);
+        } else if (_auxData == 1) {
+            if (_inputAssetA.assetType != AztecTypes.AztecAssetType.ERC20) {
+                revert ErrorLib.InvalidInputA();
+            }
+
+            if (_outputAssetA.assetType == AztecTypes.AztecAssetType.ETH) {
+                outputValueA = _unzapETH(_inputValue, _interactionNonce, _inputAssetA);
+            } else if (_outputAssetA.assetType == AztecTypes.AztecAssetType.ERC20) {
+                outputValueA = IYearnVault(_inputAssetA.erc20Address).withdraw(_inputValue);
+            } else {
+                revert ErrorLib.InvalidOutputA();
+            }
+        } else {
+            revert ErrorLib.InvalidAuxData();
+        }
+        return (outputValueA, 0, false);
+    }
+
+    /**
+     * @notice zap _inputValue ETH to wETH.
+     * @param _inputValue - Amount of underlying to deposit
+     * @param _outputAssetA - Vault we want to deposit to
+     */
+    function _zapETH(uint256 _inputValue, AztecTypes.AztecAsset memory _outputAssetA) private {
+        if (msg.value == 0 || msg.value != _inputValue) {
```

Comment:
This check is also unnecessary. You can trust that the `RollupProcessor` sends correct amounts.